src/bun.js/api/bun/subprocess.zig

```diff
@@ -2505,14 +2523,13 @@ pub const Subprocess = struct {
         //
         // When Bun.spawn() is given an `.ipc` callback, it enables IPC as follows:
         var socket: IPC.Socket = undefined;
-        if (ipc_mode != .none) {
-            if (comptime is_sync) {
-                globalThis.throwInvalidArguments("IPC is not supported in Bun.spawnSync", .{});
-                return .zero;
-            }
-
+        if ((comptime !is_sync) and ipc_mode != .none) {
             env_array.ensureUnusedCapacity(allocator, 2) catch |err| return globalThis.handleError(err, "in posix_spawn");
```

Comment:
this should be greater than 2 right? at least the length of `"NODE_CHANNEL_FD=3"` I think

Reply:
nvm lol

Reply:
number of slices not the slice content themselves. the slices are ro memory.
i will adjust this to 3 and document that the extra number is for the null at the end